packages/fxa-auth-server/scripts/cleanup-old-carts.ts

```diff
@@ -11,14 +11,28 @@ const pckg = require('../package.json');
 
 const anonymizeableFields = new Set(['taxAddress'] as const);
 
-const parseDeleteBefore = (deleteBefore: string | number) => {
+const parseDeleteBefore = (deleteBefore: string | number | null) => {
+  if (!deleteBefore) {
+    return null;
+  }
   const date = new Date(deleteBefore);
   if (!date.getTime()) {
     throw new Error('deleteBefore is invalid');
   }
   return date;
 };
 
+const parseDeleteBeforeDays = (deleteBeforeDays: string | number) => {
```

Comment:
This should have a guard similar to parseDeleteBefore, otherwise [this line](https://github.com/mozilla/fxa/pull/19190/files#diff-6e9ffa233c06710cb205c90c44f0e7a163a3108ae3e8cd469083c33ea6cd3587R111-R115) will never be reached in the case of --delete-before not being provided, since [this line](https://github.com/mozilla/fxa/pull/19190/files#diff-6e9ffa233c06710cb205c90c44f0e7a163a3108ae3e8cd469083c33ea6cd3587R30-R32) will short circuit